Combined the `in_container_utils.sh` with run_docs_build.sh

KesharwaniArpita opened the pull request:
Combined the `in_container_utils.sh with run_docs_build.sh` to eliminate the dependency of the latter on the former in relation with the issue #26019 . The file is still in bash which will be converted to `.py` files once modified and approved.

KesharwaniArpita committed:
Create run_anything.py

this file contains the contents of the run_anything.sh file rewritten in python

KesharwaniArpita committed:
Merge branch 'apache:main' into bashtopy

KesharwaniArpita committed:
Merge branch 'apache:main' into bashtopy

KesharwaniArpita committed:
Combined the `_init_container_script_init.sh and run_docs_build.sh

Combining the functionalities of `_init_container_script_init.sh` and `run_docs_build.sh` to remove the dependency of the latter on the former.

KesharwaniArpita committed:
Merge pull request #3 from KesharwaniArpita/KesharwaniArpita-patch-1

Combined the `_init_container_script_init.sh and run_docs_build.sh

KesharwaniArpita committed:
Combined the `in_container_utils.sh with run_docs_build.sh

Combined the `in_container_utils.sh with run_docs_build.sh to eliminate the dependency of the latter on the former.

KesharwaniArpita committed:
Merge pull request #4 from KesharwaniArpita/KesharwaniArpita-patch-1

Combined the `in_container_utils.sh with run_docs_build.sh

KesharwaniArpita committed:
Merge branch 'apache:main' into bashtopy

potiuk reviewed (changes requested):
It's pretty much not what the inteion of rewriting the bash scripts is about

KesharwaniArpita committed:
Merge branch 'apache:main' into bashtopy

KesharwaniArpita committed:
Merge branch 'apache:main' into bashtopy

KesharwaniArpita committed:
integrating the necessary functionality from the run_init_script.sh file into the startairflow_if_requested() function in check_environment.sh

Removing the dependency on the run_init_script.sh file and integrating its functionality directly into the startairflow_if_requested() function to streamline the script and reduce dependencies.

KesharwaniArpita committed:
Merge pull request #6 from KesharwaniArpita/KesharwaniArpita-patch-1

integrating the necessary functionality from the run_init_script.sh f…

potiuk commented on scripts/in_container/run_docs_build.sh:
```diff
@@ -16,12 +16,64 @@
 # under the License.
 # shellcheck source=scripts/in_container/_in_container_script_init.sh
-. "$( dirname "${BASH_SOURCE[0]}" )/_in_container_script_init.sh"
+
```
Almost all of it can be removed as it is not used + there are some left-overs (like shellcheck above) that are not needed any more

potiuk commented on scripts/in_container/run_anything.py:
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
```
Do we need that?

potiuk commented on scripts/in_container/check_environment.sh:
```diff
@@ -112,10 +112,32 @@ function startairflow_if_requested() {
         export AIRFLOW__CORE__LOAD_EXAMPLES=${LOAD_EXAMPLES}
 
-        . "$( dirname "${BASH_SOURCE[0]}" )/configure_environment.sh"
+        if [ -z "${FILES_DIR+x}" ]; then
```
I am not sure if in this case it's worth incorporating check environment in. What I really meant to is ratheer to rewrite the configure_environment.sh and run it from here. It makes little sense to increase duplication of the check_environment.sh into other scripts

KesharwaniArpita committed:
Merge branch 'apache:main' into bashtopy

KesharwaniArpita committed:
Refactor check_environment.sh Integration in Script

This PR addresses the concerns related to the inclusion of check_environment.sh in the script. As per the feedback, this update rewrites the configure_environment.sh and incorporates it directly into the script. This change eliminates the redundancy of including the check_environment.sh in multiple scripts, enhancing code cleanliness and maintainability. The modification aims to streamline the execution process and improve the overall efficiency of the script.

KesharwaniArpita committed:
Merge branch 'apache:main' into bashtopy

KesharwaniArpita committed:
Refactor Docker Documentation Build Script run_docs_build.sh

This pull request refactors the Docker documentation build script in the Airflow project. The changes remove unused elements such as _in_container_script_init.sh and obsolete shellcheck references. The script has been streamlined for improved readability and maintenance. Additionally, this update ensures that the script remains focused on its core functionality, enhancing the overall efficiency of the build process.

This modification aligns with the goal of reducing unnecessary elements in the Airflow scripts, contributing to a more maintainable and cohesive codebase.

amoghrajesh reviewed (changes requested):
I am a little confused with the intention of this PR. The title vs the changes makes it confusing. 

From what I understand, the task was to rewrite: check_environment.sh into python. 
Instead of changing multiple files at once, we should go small and step by step in my opinion. Make changes to check_environment.sh for instance and test it well.
